src/main/java/com/khutircraftubackend/auth/UserService.java

```diff
@@ -0,0 +1,24 @@
+package com.khutircraftubackend.auth;
+
+import com.khutircraftubackend.exception.user.UserNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.security.Principal;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public UserEntity findUserForPrincipal(Principal principal) {
```

Comment:
naming convention for methods in JPA: 
https://docs.spring.io/spring-data/jpa/reference/jpa/query-methods.html 
findUserForPrincipal &#8594; findByPrincipal